File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic.Syntax/XmlPrefixSyntax.cs
using Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax;

namespace Microsoft.CodeAnalysis.VisualBasic.Syntax
{
	public sealed class XmlPrefixSyntax : VisualBasicSyntaxNode
	{
		public SyntaxToken Name => new SyntaxToken(this, ((Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.XmlPrefixSyntax)base.Green)._name, base.Position, 0);

		public SyntaxToken ColonToken => new SyntaxToken(this, ((Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.XmlPrefixSyntax)base.Green)._colonToken, GetChildPosition(1), GetChildIndex(1));

		internal XmlPrefixSyntax(GreenNode green, SyntaxNode parent, int startLocation)
			: base(green, parent, startLocation)
		{
		}

		internal XmlPrefixSyntax(SyntaxKind kind, DiagnosticInfo[] errors, SyntaxAnnotation[] annotations, XmlNameTokenSyntax name, PunctuationSyntax colonToken)
			: this(new Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.XmlPrefixSyntax(kind, errors, annotations, name, colonToken), null, 0)
		{
		}

		public XmlPrefixSyntax WithName(SyntaxToken name)
		{
			return Update(name, ColonToken);
		}

		public XmlPrefixSyntax WithColonToken(SyntaxToken colonToken)
		{
			return Update(Name, colonToken);
		}

		internal override SyntaxNode GetCachedSlot(int i)
		{
			return null;
		}

		internal override SyntaxNode GetNodeSlot(int i)
		{
			return null;
		}

		public override TResult Accept<TResult>(VisualBasicSyntaxVisitor<TResult> visitor)
		{
			return visitor.VisitXmlPrefix(this);
		}

		public override void Accept(VisualBasicSyntaxVisitor visitor)
		{
			visitor.VisitXmlPrefix(this);
		}

		public XmlPrefixSyntax Update(SyntaxToken name, SyntaxToken colonToken)
		{
			if (name != Name || colonToken != ColonToken)
			{
				XmlPrefixSyntax xmlPrefixSyntax = SyntaxFactory.XmlPrefix(name, colonToken);
				SyntaxAnnotation[] annotations = GetAnnotations();
				if (annotations != null && annotations.Length > 0)
				{
					return SyntaxNodeExtensions.WithAnnotations(xmlPrefixSyntax, annotations);
				}
				return xmlPrefixSyntax;
			}
			return this;
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/Syntax/XmlPrefixSyntax.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic
Imports Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
Imports System

Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax
	Public NotInheritable Class XmlPrefixSyntax
		Inherits Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxNode
		Public ReadOnly Property ColonToken As Microsoft.CodeAnalysis.SyntaxToken
			Get
				Return New Microsoft.CodeAnalysis.SyntaxToken(Me, DirectCast(MyBase.Green, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.XmlPrefixSyntax)._colonToken, Me.GetChildPosition(1), MyBase.GetChildIndex(1))
			End Get
		End Property

		Public ReadOnly Property Name As Microsoft.CodeAnalysis.SyntaxToken
			Get
				Return New Microsoft.CodeAnalysis.SyntaxToken(Me, DirectCast(MyBase.Green, Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.XmlPrefixSyntax)._name, MyBase.Position, 0)
			End Get
		End Property

		Friend Sub New(ByVal green As GreenNode, ByVal parent As SyntaxNode, ByVal startLocation As Integer)
			MyBase.New(green, parent, startLocation)
		End Sub

		Friend Sub New(ByVal kind As SyntaxKind, ByVal errors As DiagnosticInfo(), ByVal annotations As SyntaxAnnotation(), ByVal name As XmlNameTokenSyntax, ByVal colonToken As PunctuationSyntax)
			MyClass.New(New Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax.XmlPrefixSyntax(kind, errors, annotations, name, colonToken), Nothing, 0)
		End Sub

		Public Overrides Function Accept(Of TResult)(ByVal visitor As VisualBasicSyntaxVisitor(Of TResult)) As TResult
			Return visitor.VisitXmlPrefix(Me)
		End Function

		Public Overrides Sub Accept(ByVal visitor As Microsoft.CodeAnalysis.VisualBasic.VisualBasicSyntaxVisitor)
			visitor.VisitXmlPrefix(Me)
		End Sub

		Friend Overrides Function GetCachedSlot(ByVal i As Integer) As SyntaxNode
			Return Nothing
		End Function

		Friend Overrides Function GetNodeSlot(ByVal i As Integer) As SyntaxNode
			Return Nothing
		End Function

		Public Function Update(ByVal name As Microsoft.CodeAnalysis.SyntaxToken, ByVal colonToken As Microsoft.CodeAnalysis.SyntaxToken) As Microsoft.CodeAnalysis.VisualBasic.Syntax.XmlPrefixSyntax
			Dim xmlPrefixSyntax As Microsoft.CodeAnalysis.VisualBasic.Syntax.XmlPrefixSyntax
			If (name <> Me.Name OrElse colonToken <> Me.ColonToken) Then
				Dim xmlPrefixSyntax1 As Microsoft.CodeAnalysis.VisualBasic.Syntax.XmlPrefixSyntax = Microsoft.CodeAnalysis.VisualBasic.SyntaxFactory.XmlPrefix(name, colonToken)
				Dim annotations As SyntaxAnnotation() = MyBase.GetAnnotations()
				xmlPrefixSyntax = If(annotations Is Nothing OrElse CInt(annotations.Length) <= 0, xmlPrefixSyntax1, xmlPrefixSyntax1.WithAnnotations(annotations))
			Else
				xmlPrefixSyntax = Me
			End If
			Return xmlPrefixSyntax
		End Function

		Public Function WithColonToken(ByVal colonToken As Microsoft.CodeAnalysis.SyntaxToken) As Microsoft.CodeAnalysis.VisualBasic.Syntax.XmlPrefixSyntax
			Return Me.Update(Me.Name, colonToken)
		End Function

		Public Function WithName(ByVal name As Microsoft.CodeAnalysis.SyntaxToken) As Microsoft.CodeAnalysis.VisualBasic.Syntax.XmlPrefixSyntax
			Return Me.Update(name, Me.ColonToken)
		End Function
	End Class
End Namespace